src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj

```diff
@@ -63,175 +62,19 @@
-    </ItemGroup>
-    <Message Text="$(MSBuildProjectName) -&gt; $(PathToMergedNuGetPackResource)" Importance="High" />
-  </Target>
-
-  <Target Name="ILMergeNuGetPackResources"
-          DependsOnTargets="GetILMergePlatformArg;DetermineILMergeNuGetPackResourcesInputsOutputs"
-          Inputs="$(PathToBuiltNuGetPackResource);@(ResourceArtifacts)"
-          Outputs="$(PathToMergedNuGetPackResource)"
-          Condition="'$(TargetFramework)' != '' AND '$(CurrentCulture)' != ''">
-    <PropertyGroup>
-      <ILMergeResultDir>$([System.IO.Path]::GetDirectoryName($(PathToMergedNuGetPackResource)))</ILMergeResultDir>
-    </PropertyGroup>
-
-    <MakeDir Directories="$(ILMergeResultDir)" />
-
-    <PropertyGroup>
-      <IlmergeCommand>$(ILMergeConsolePath) ^
-        /out:$(PathToMergedNuGetPackResource) ^
-        $(ILMergePlatformArg) ^
-        /allowDup ^
-        /log ^
-        /lib:$(OutputPath)\publish ^
-        /internalize ^
-        /xmldocs ^
-        /log:$(ILMergeResultDir)IlMergeLogResources.txt</IlmergeCommand>
-      <IlmergeCommand Condition="'$(SkipSigning)' != 'true' And Exists($(AssemblyOriginatorKeyFile))">$(IlmergeCommand) ^
-        /delaysign ^
-        /keyfile:$(AssemblyOriginatorKeyFile)</IlmergeCommand>
-      <IlmergeCommand>$(IlmergeCommand) ^
-        $(PathToBuiltNuGetPackResource) ^
-        @(ResourceArtifacts, ' ')</IlmergeCommand>
-    </PropertyGroup>
-    <Exec Command="$(IlmergeCommand)" ContinueOnError="false" />
-  </Target>
-
-  <!--These targets help get all the DLLs that are packed in the nuspec to be sent for signing.-->
-  <Target Name="GetIlmergeBuildOutput" DependsOnTargets="ILMergeNuGetPackResourcesPerCulture;_GetDllsInOutputDirectory" Returns="@(DllsToSign)">
-    <ItemGroup>
-      <DllsToSign Include="@(DllsInOutputDir)" KeepDuplicates="false" />
-    </ItemGroup>
-  </Target>
-
-  <Target Name="_GetDllsInOutputDirectory" Returns="@(DllsInOutputDir)">
-    <ItemGroup>
-      <DllsInOutputDir Include="$(OutputPath)ilmerge\*.dll" KeepDuplicates="false">
-        <StrongName>MsSharedLib72</StrongName>
-        <Authenticode>Microsoft400</Authenticode>
-      </DllsInOutputDir>
-      <DllsInOutputDir Include="$(OutputPath)ilmerge\*.xml" KeepDuplicates="false">
-        <Authenticode>MicrosoftXmlSHA2</Authenticode>
-      </DllsInOutputDir>
-    </ItemGroup>
-  </Target>
-
-  <!--These targets help get all the DLLs that are packed in the nuspec to be sent for signing.-->
-  <Target Name="GetIlmergeSymbolOutput" Returns="@(SymbolsToIndex)">
-    <ItemGroup>
-      <SymbolsToIndex Include="$(OutputPath)ilmerge\*.pdb" KeepDuplicates="false" />
-    </ItemGroup>
-  </Target>
-
-  <Target Name="GetFinalBuildOutput" Returns="@(DllsToIndex)">
-    <ItemGroup>
-      <DllsToIndex Include="$(OutputPath)ilmerge\NuGet.Build.Tasks.Pack.dll"/>
-    </ItemGroup>
-  </Target>
-
   <Target Name="CreatePackNupkg">
```

Comment:
I think you could just get rid of this altogether, and use stock packaging. It would require some downstream reaction.

Reply:
I have a follow-up in https://github.com/NuGet/Home/blob/dev/accepted/2025/consolidate-pack-and-restore.md. 

We need to tweak the .NET SDK layout to allow pack to load the framework tasks for pack so that'll be the next step.